### PR TITLE
Ported to typsecript

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Ported repo to typescript. Fixed dependencies and updated scripts. Also made minor changes to test and api. Issued out implementation error.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,16 @@
   "name": "platform-chaos-api",
```
Comment (bengreenier) [inline package.json]: i'll assume it's implicit that the comments on the other, similar ts pr you've been working on will get applied here

File: src/lib/app.ts [hunk 2]
```diff
@@ -17,6 +18,7 @@ module.exports = (opts) => {
     fsLocation: './webapp-extension-store.json'
   })
   azureChaos.factory.Logger.configure({
+    // tslint:disable-next-line
```
Comment (bengreenier) [inline src/lib/app.ts]: this is a fun one

File: src/lib/app.ts [hunk 3]
```diff
@@ -30,18 +32,18 @@ module.exports = (opts) => {
   if (opts && opts.isProd) {
     // in production, use Bearer Strategy for auth
     passport.use(new BearerStrategy({
-      identityMetadata: 'https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration',
-      clientID: opts.authClientId,
       audience: opts.authAudience,
+      clientID: opts.authClientId,
+      identityMetadata: 'https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration',
```
Comment (bengreenier) [inline src/lib/app.ts]: this should come from opts, and default to this value

File: src/lib/app.ts [hunk 4]
```diff
@@ -30,18 +32,18 @@ module.exports = (opts) => {
...
       issuer: opts.authIssuer,
       loggingLevel: 'error'
```
Comment (bengreenier) [inline src/lib/app.ts]: this should come from opts and default to this value

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (bengreenier) [inline tsconfig.json]: i'll assume it's implicit that the comments on the other, similar ts pr you've been working on will get applied here

File: src/lib/app.ts (hunk 3 above)
Comment (Ethan-Arrowood) [reply src/lib/app.ts]: IMO we should move this concern to issue #17

File: src/lib/app.ts (hunk 4 above)
Comment (Ethan-Arrowood) [reply src/lib/app.ts]: IMO we should move this concern to issue #17

**Pull request merged**